File: 78. Subsets.swift
class Solution {
    
    var result: Array<Array<Int>> = []
    var nums: Array<Int> = []
    
    var array: Array<Int> = []
    
    func backtrackingDFS(_ index: Int) {
        
        if index >= nums.count {
            result.append(array)
            return
        }
        
        array.append(nums[index])
        
        // Use The current element
        backtrackingDFS(index+1)
		
		// Backtrack
        array.popLast()
        backtrackingDFS(index+1)
        
    }
    
    func subsets(_ nums: Array<Int>) -> Array<Array<Int>> {
        self.nums = nums
        backtrackingDFS(0)
        return result
    }
}

File: 78. Subsets.py
from typing import List

class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        result = []
        def backtrackingDFS(index: int,curArray: List[int]):
            if index >= len(nums):
                result.append(curArray[:])
                return
            # Add Element To Array
            curArray.append(nums[index])
            backtrackingDFS(index+1,curArray)
            # Backtrack
            curArray.pop()
            backtrackingDFS(index+1,curArray)
        backtrackingDFS(0,[])
        return result